pkg/netpol/eval/internal/k8s/adminnetpol.go

```diff
@@ -17,12 +18,183 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 	apisv1a "sigs.k8s.io/network-policy-api/apis/v1alpha1"
 
+	"github.com/np-guard/models/pkg/netset"
+
 	pkgcommmon "github.com/np-guard/netpol-analyzer/pkg/internal/common"
 	"github.com/np-guard/netpol-analyzer/pkg/internal/netpolerrors"
+	"github.com/np-guard/netpol-analyzer/pkg/logger"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/alerts"
 	"github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
 )
 
-//// first section in the file contains:
+// AdminNetworkPolicy is an alias for k8s adminNetworkPolicy object
+type AdminNetworkPolicy struct {
+	*apisv1a.AdminNetworkPolicy                 // embedding k8s admin-network-policy object
+	warnings                    map[string]bool // set of warnings which are raised by the anp
+}
+
+// note that could not use Generics with GO 1.21 or older versions; since:
+// according to https://tip.golang.org/doc/go1.18#generics :
+// "The Go compiler does not support accessing a struct field x.f where x is of type parameter type even if all types in the type
+// parameter’s type set have a field f. We may remove this restriction in a future release."
+// (till GO 1.21 this restriction is not removed yet.)
+// and to resolve remaining duplicated code for AdminNetworkPolicy and BaselineAdminNetworkPolicy we need the option of using
+// the inner fields of  generic type in the funcs, either implicitly or explicitly.
+// @todo: with upgraded GO version, check if using generics may help avoid remaining duplicates in
+// the files adminnetpol.go and baseline_admin_netpol.go
```

Comment:
this comment can be removed from here, and only remain in the issue opened for this.